dispatch.go

```diff
@@ -40,13 +40,20 @@ func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res i
 	profiler := d.log.StartProfiler("call %s", name)
 	defer profiler.Stop()
 
+	select {
+	case <-d.stopCh:
```

Comment:
Actually, this isn't needed for correctness, right? Since if we're closed we'll hit the stopCh path below or the errCh path. If so, I'd rather just remove this, as it muddies the waters.